androidClient/app/src/main/java/com/macbitsgoa/comrades/coursematerial/CourseActivity.java

```diff
@@ -110,72 +110,72 @@ protected void onCreate(Bundle savedInstanceState) {
         btnAddMaterial.setOnClickListener(this);
     }
 
-    private void receiveDownloadMessage() {
+    private void initBroadCastReceiver() {
         broadcastReceiver = new BroadcastReceiver() {
             @Override
-            public void onReceive(Context context, Intent intent) {
-                Bundle resultData = intent.getExtras();
+            public void onReceive(final Context context, final Intent intent) {
+                final Bundle resultData = intent.getExtras();
                 if (resultData != null) {
-                    String itemId = resultData.getString("id");
-                    int resultCode = resultData.getInt("resultCode");
+                    final String itemId = resultData.getString("id");
+                    if (itemId == null) {
+                        Log.e(TAG, "itemId was found null. Returning", new Throwable("Trace").fillInStackTrace());
+                        return;
+                    }
+                    final int resultCode = resultData.getInt("resultCode");
+                    final int index = CourseMaterial.findIndex(materialArrayList, itemId);
```

Comment:
added a single utility function (in `CourseMaterial`) to remove code duplication